hdf5-vol-daos.spec

```diff
@@ -0,0 +1,174 @@
+Source0: daos-vol-master@b324b90d455.zip
+Source1: vol-tests-master@31660ab1935.zip
+
+%if 0%{?suse_version}
+BuildRequires: gcc-fortran
+BuildRequires: lua-lmod
+BuildRequires: gcc, gcc-c++
+%endif
+BuildRequires: cmake
+BuildRequires: hdf5-devel%{?_isa}
+
+%description
+HDF5 VOL DAOS connector is used to leverage the
+capabilities of the DAOS object storage system
+within an HDF5 application.  It translates HDF5 VOL 
+storage related calls into native daos storage operations
+
+
+%if %{with_mpich}
+%package mpich
+Summary: HDF5 Vol Daos for MPICH
+BuildRequires: hdf5-mpich-devel%{?_isa}
+BuildRequires: mpich-devel%{?_isa}
+Requires: %{name}%{?_isa} = %{version}-%{release}
+
+%description mpich
+HDF5 Vol Daos for MPICH
+
+%package mpich-tests
+Summary: HDF5 Vol Daos mpich tests
+BuildRequires: hdf5-mpich-devel%{?_isa}
+BuildRequires: mpich-devel%{?_isa}
+Requires: %{name}%{?_isa} = %{version}-%{release}
+
+%description mpich-tests
+HDF5 Vol Daos mpich tests
+
+%endif
+
+%if %{with_openmpi3}
+%package openmpi3
+Summary: HDF5 Vol Daos  OpenMPI 3
+BuildRequires: hdf5-openmpi3-devel%{?_isa}
+BuildRequires: openmpi3-devel%{?_isa}
+Requires: %{name}%{?_isa} = %{version}-%{release}
+
+%description openmpi3
+HDF5 Vol Daos  OpenMPI 3
+
+%package openmpi3-tests
+Summary: HDF5 Vol Daos openmpi3 tests
+BuildRequires: hdf5-openmpi3-devel%{?_isa}
+BuildRequires: openmpi3-devel%{?_isa}
+Requires: %{name}%{?_isa} = %{version}-%{release}
+
+%description openmpi3-tests
+HDF5 Vol Daos openmpi3 tests
+%endif
+
+%prep
+%setup -c
+cd test/vol
+/usr/bin/unzip %SOURCE1
+
+%build
+mkdir build
+pushd build
+cmake -DCMAKE_INSTALL_PREFIX=%{_prefix} \
+      -DBUILD_TESTING=ON \
+      -DHDF5_VOL_DAOS_INSTALL_LIB_DIR=%{_libdir} \
+      -DHDF5_VOL_TEST_ENABLE_PART=ON \
+      -DHDF5_VOL_TEST_ENABLE_PARALLEL=ON \
+      -DHDF5_VOL_DAOS_USE_SYSTEM_HDF5=OFF \
+      -DDAOS_UNIFIED_NAMESPACE_LIBRARY=%{_libdir}/libduns.so \
+      ..
+%{make_build}
+popd
+for mpi in %{?mpi_list}
+do
+  mkdir $mpi
+  pushd $mpi
+  %module_load $mpi
+  cmake -DCMAKE_INSTALL_PREFIX=%{_libdir}/$mpi \
+        -DBUILD_TESTING=ON \
+        -DHDF5_VOL_TEST_ENABLE_PART=ON \
+        -DHDF5_VOL_TEST_ENABLE_PARALLEL=ON \
+        -DHDF5_VOL_DAOS_USE_SYSTEM_HDF5=OFF \
+        -DDAOS_UNIFIED_NAMESPACE_LIBRARY=%{_libdir}/libduns.so \
+        -DMPI_C_COMPILER=%{_libdir}/$mpi/bin/mpicc \
+        ..
+  %{make_build}
+  module purge
+  popd
+done
+
+%install
+%{make_install} -C build
+
+for mpi in %{?mpi_list}
+do
+  %module_load $mpi
+  %{make_install} -C $mpi
+  module purge
+done
+
+# install tests
+for mpi in %{?mpi_list}
+do
+  mkdir -p ${RPM_BUILD_ROOT}%{_libdir}/hdf5_vol_daos/$mpi/tests/
+  for x in h5_test_testhdf5 h5vl_test h5_partest_t_bigio h5_partest_testphdf5 \
+           h5vl_test_parallel h5_partest_t_shapesame
+  do
+    install -m 0755 $mpi/bin/${x} ${RPM_BUILD_ROOT}%{_libdir}/hdf5_vol_daos/$mpi/tests/
+  done
+done
+
+%files
+%license COPYING
+%{_libdir}/libhdf5_vol_daos.so*
```

Comment:
Is this including the `.so` _development_ file?  Shouldn't that be in a `-devel` subpackage?  What about the `/usr/include` files that the MPI builds seem to be including in their package?  Those should also go in a `-devel` also.